lib/src/rules/prefer_null_aware_operator.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/token.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:linter/src/analyzer.dart';
+
+const _desc = r'Prefer null aware operator.';
+
+const _details = r'''
+
+Prefer null aware operator instead of null check in conditional expression.
+
+**BAD:**
+```
+v = a == null ? null : a.b;
+```
+
+**GOOD:**
+```
+v = a?.b;
+```
+
+''';
+
+class PreferNullAwareOperator extends LintRule implements NodeLintRule {
+  PreferNullAwareOperator()
+      : super(
+            name: 'prefer_null_aware_operator',
+            description: _desc,
+            details: _details,
+            group: Group.style);
+
+  @override
+  void registerNodeProcessors(NodeLintRegistry registry,
+      [LinterContext context]) {
+    final visitor = new _Visitor(this);
+    registry.addConditionalExpression(this, visitor);
+  }
+}
+
+class _Visitor extends SimpleAstVisitor {
+  final LintRule rule;
+
+  _Visitor(this.rule);
+
+  @override
+  visitConditionalExpression(ConditionalExpression node) {
+    final condition = node.condition;
+    if (condition is BinaryExpression &&
+        (condition.operator.type == TokenType.EQ_EQ ||
+            condition.operator.type == TokenType.BANG_EQ)) {
+      // ensure pattern `xx == null ? null : yy` or `xx != null ? yy : null`
+      if (condition.operator.type == TokenType.EQ_EQ) {
+        if (node.thenExpression is! NullLiteral) {
+          return;
+        }
+      } else {
+        if (node.elseExpression is! NullLiteral) {
+          return;
+        }
+      }
+
+      // ensure condition is a null check
+      Expression expression;
+      if (condition.leftOperand is NullLiteral) {
+        expression = condition.rightOperand;
+      } else if (condition.rightOperand is NullLiteral) {
+        expression = condition.leftOperand;
+      }
+
+      // lint only if local variable is subject to avoid side effects
+      if (expression != null &&
```

Comment:
nit: The null comparison is not necessary because of the `is` check on the next line.

Reply:
done